### PR TITLE
Fix Overclock Calculator Making Too many parallels and wrong eut consumption

### PR DESCRIPTION
**Pull request by BlueWeabo**
Thanks to @GDCloudstrike I managed to fix the bugs before it became too wide spread

### CODE REVIEW

File: src/main/java/gregtech/api/util/GT_OverclockCalculator.java [hunk 1]
```diff
@@ -549,13 +566,42 @@ public double calculateDurationUnderOneTick() {
      * @param originalMaxParallel Parallels which are of the actual machine before the overclocking extra ones
      */
     public long calculateEUtConsumptionUnderOneTick(int originalMaxParallel, int currentParallel) {
-        double parallelOverclocks = Math.log((double) currentParallel / originalMaxParallel)
-            / Math.log(1 << durationDecreasePerOC);
-        return (long) Math.floor(
-            recipeVoltage * Math.pow(1 << eutIncreasePerOC, parallelOverclocks)
+        double heatDiscountMultiplier = calculateHeatDiscountMultiplier();
+        calculateFinalRecipeEUt(heatDiscountMultiplier);
```
Comment (miozune) [inline src/main/java/gregtech/api/util/GT_OverclockCalculator.java:570]: This does nothing

File: src/main/java/gregtech/api/util/GT_OverclockCalculator.java [hunk 2]
```diff
@@ -549,13 +566,42 @@ public double calculateDurationUnderOneTick() {
...
+        // anomalies.
+        // = Everything else for recipe voltage is also calculated here.
+
+        double parallelMultiplierFromOverclocks = (double) currentParallel / originalMaxParallel;
+        double amountOfParallelHeatOverclocks = Math.min(
+            Math.log(parallelMultiplierFromOverclocks) / Math.log(1 << durationDecreasePerHeatOC),
+            calculateAmountOfHeatOverclocks());
+        double amountOfParallelOverclocks = Math.log(parallelMultiplierFromOverclocks)
+            / Math.log(1 << durationDecreasePerOC)
+            - amountOfParallelHeatOverclocks * ((1 << durationDecreasePerHeatOC) / (1 << durationDecreasePerOC));
```
Comment (miozune) [inline src/main/java/gregtech/api/util/GT_OverclockCalculator.java:593]: `((1 << durationDecreasePerHeatOC) / (1 << durationDecreasePerOC))` can be `(1 << (durationDecreasePerHeatOC - durationDecreasePerOC))`

**Review by miozune — COMMENTED** [review]
So, is it possible to remove `GT_OverclockCalculator(@Nonnull GT_OverclockCalculator calculator)` since it's unused?

**Review by miozune — APPROVED** [review]
Thanks for the thorough documentation!

**Comment by BlueWeabo** [comment]
Yeah I needed it for myself tooo. Things got complicated fast.

**Pull request merged**